tests/bluetooth/controller/mock_ctrl/include/hal/ticker_vendor_hal.h

```diff
@@ -29,6 +29,16 @@
 		HAL_TICKER_CNTR_MASK \
 	)
 
+/* Macro to translate microseconds to tick units.
+ * NOTE: This returns the ceil value.
+ */
+#define HAL_TICKER_US_TO_TICKS_CEIL(x) \
+	( \
+		((uint32_t)(((((uint64_t) (x) * 1000000000UL) + \
+			      HAL_TICKER_CNTR_CLK_UNIT_FS - 1U)) / \
+			    HAL_TICKER_CNTR_CLK_UNIT_FS)) & \
+		HAL_TICKER_CNTR_MASK \
+	)
```

Comment:
Which of the value(s) are vendor specific, since we need several versions of this macro? 

Could we introduce a generic macro and simply have the vendor specific value as input?

Reply:
The macro calculation is vendor specific. 
For platforms where one tick is one microsecond, then `HAL_TICKER_US_TO_TICKS_CEIL` would just be `(x)`. Hence, all the HAL_TICKER_* is currently vendor specific.

Reply:
> For platforms where one tick is one microsecond, then HAL_TICKER_US_TO_TICKS_CEIL would just be (x). 

But in that could, couldn't we just do

```c
#define HAL_TICKER_US_TO_TICKS_CEIL(x, factor) \
	( \
		((uint32_t)(((((uint64_t) (x) * (factor)) + \
			      HAL_TICKER_CNTR_CLK_UNIT_FS - 1U)) / \
			    HAL_TICKER_CNTR_CLK_UNIT_FS)) & \
		HAL_TICKER_CNTR_MASK \
	)
```

Reply:
On nRF calculation is specific to have integer arithmetic when the clock is 32768 Hz, i.e. (1/32768)*1000000000 = 30517578125.

But say the clock is 16 MHz, then the calculation is simply ((x) * 16), it is cleaner to have separate hal/ticker.h file for different platforms with differing clock frequencies.

Reply:
Fair point - No reason to change in this PR anyways :)